src/Entity/Zone.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace App\Entity;
+
+use App\Entity\Behaviour\EntityHistoryInterface;
+use App\Entity\Enum\HistoryEntryEvent;
+use App\Repository\ZoneRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Validator\Constraints as Assert;
+
+#[ORM\Entity(repositoryClass: ZoneRepository::class)]
+#[ORM\UniqueConstraint(columns: ['name', 'territory_id'])]
+#[UniqueEntity(
+    fields: ['name', 'territory'],
+    message: 'Ce nom est déjà utilisé sur le territoire.',
+    errorPath: 'name',
+)]
+class Zone implements EntityHistoryInterface
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\Column(type: Types::TEXT)]
+    private ?string $area = null;
+
+    #[ORM\ManyToOne(inversedBy: 'zones')]
+    #[ORM\JoinColumn(nullable: false)]
```

Comment:
par défaut c'est false, par exemple c'est pas mentionné sur `name` mais le champs est bien à not null

Reply:
Non c'est le cas par défaut pour les champs classiques mais apparemment pas pour les relations : 
![Capture d’écran 2024-10-23 111906](https://github.com/user-attachments/assets/db7796cc-f20d-4a1b-a8aa-9688ee8e1b70)

Quoi qu'il en soit je propose de garder ce que génère la commande `make:entity` de Symfony. Et je propose de faire pareil pour la valeur de retour généré des setter `: static` qu'on oubli toujours de passer en `static`, donc a une codebase qui mélange les deux, autant dire que maintenant on garde ce qui est généré (vu que ca n'a aucun impact)